apps/dashboard/src/components/workflow-editor/test-workflow/test-workflow-logs-sidebar.tsx

```diff
@@ -37,8 +42,12 @@ export const TestWorkflowLogsSidebar = ({ transactionId, workflow }: TestWorkflo
       refetchInterval: shouldRefetch ? 1000 : false,
     }
   );
-  const activity = activities?.[0];
-  const activityId: string | undefined = parentActivityId ?? activity?._id;
+
+  const activityId: string | undefined = parentActivityId ?? activities?.[0]?._id;
+  const { activity: latestActivity, isPending: isActivityPending, error: activityError } = usePullActivity(activityId);
+  const activity = latestActivity ?? activities?.[0];
+  const isPending = areActivitiesPending || isActivityPending;
+  const error = activitiesError || activityError;
```

Comment:
First, we fetch the activities by the `transactionId`, then, when there are "activities" created, we start the single activity pooling. 
In the case when the activity gets merged to the digested one, we do fetch its parent details.